doc/dev/python_scheduling/interactive_scheduling_layer_norm.py

```diff
@@ -0,0 +1,190 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# Owner(s): ["module: nvfuser"]
+
+# In[1]:
+
+
+import torch
+from nvfuser import (
+    FusionDefinition,
+    DataType,
+    ParallelType,
+    MemoryType,
+)
+
+
+# In[2]:
+
+
+batch_size = 1024
+tensor_size = 4096
+inputs = [
+    torch.randn(batch_size, tensor_size, dtype=torch.bfloat16, device="cuda"),
+]
+
+
+# In[3]:
+
+
+class LayerNorm(FusionDefinition):
+    def definition(self):
+        self.t0 = self.from_pytorch(inputs[0])
+        self.s0 = self.define_scalar(1e-6, dtype=DataType.Double)
+        self.norm_const = self.define_scalar(tensor_size, dtype=DataType.Int)
+
+        self.mean_cast = self.ops.cast(self.t0, dtype=DataType.Float)
+        self.sum0 = self.ops.sum(self.mean_cast, dims=[-1])
+        # NOTE Manually broadcast because fusion definition cannot access hidden reduction tensor view.
+        self.bcast_sum0 = self.ops.broadcast(self.sum0, [False, True])
+        self.mean = self.ops.div(self.bcast_sum0, self.norm_const)
+
+        self.var_cast = self.ops.cast(self.t0, dtype=DataType.Float)
+        self.diff = self.ops.sub(self.var_cast, self.mean)
+        self.diff_sq = self.ops.mul(self.diff, self.diff)
+        self.sum1 = self.ops.sum(self.diff_sq, dims=[-1])
+        # NOTE Manually broadcast because fusion definition cannot access hidden reduction tensor view.
+        self.bcast_sum1 = self.ops.broadcast(self.sum1, [False, True])
+        self.var = self.ops.div(self.bcast_sum1, self.norm_const)
+
+        self.t0_cast = self.ops.cast(self.t0, dtype=DataType.Float)
+        self.t0_diff = self.ops.sub(self.t0_cast, self.mean)
+        self.var_eps = self.ops.sqrt(self.ops.add(self.var, self.s0))
+        self.t0_norm = self.ops.div(self.t0_diff, self.var_eps)
+
+        self.t0_norm_cast = self.ops.cast(self.t0_norm, dtype=DataType.BFloat16)
+        self.add_output(self.t0_norm_cast)
+
+
+# In[4]:
+
+
+# Build FusionDefinition
+fn = LayerNorm()
+fn._setup_definition()
+fn.definition()
+fn._finalize_definition()
+
+
+# In[5]:
+
+
+# Create user schedule for this input
+# NOTE: Schedules defined by the user for specific input sizes.
+fn._setup_schedule(inputs)
+print(fn._user_schedule_ir())
+
+
+# In[6]:
```

Comment:
Is this how we want to expose our manual schedule in a tutorial?

The existing pattern is to have schedule in the derived class's `schedule` function. i.e. line 80-136 should be moved there.

Reply:
^^^ This is meant as a question rather than a suggestion. i.e. what's the reason that we are promoting a different pattern here? Does that mean we should move towards a different approach for scheduling?

cc'ing @kevinstephano who might have some comment on this.

Reply:
I imagined using a Jupyter notebook for interactive scheduling. The standard pattern forces you to rerun all cells if you want to make a change.

Reply:
keep nitpicking on this one. I would argue it probably makes sense to have a `proper` tutorial showing our intended workflow with a single `schedule()` function. :stuck_out_tongue_winking_eye: 

But we don't have to do that in this PR.

Reply:
I created `tutorial_scheduling_layer_norm.py` which uses the `schedule()` function.